messenger/http_transporter.go

```diff
@@ -85,10 +85,10 @@ func (s *stoppedState) Recv() (*Message, error)                        { return
 func (s *stoppedState) Inject(ctx context.Context, msg *Message) error { return errTerminal }
 func (s *stoppedState) Stop(graceful bool) error                       { return errTerminal }
 func (s *stoppedState) Install(messageName string)                     {}
-func (s *stoppedState) Start() <-chan error {
+func (s *stoppedState) Start() (upid.UPID, <-chan error) {
 	ch := make(chan error, 1)
 	ch <- errTerminal
-	return ch
+	return upid.UPID{}, ch
```

Comment:
We don't want to return a valid UPID in the case of an error condition.

On Thu, Sep 24, 2015 at 5:28 AM, Tomás Senart <notifications@github.com>
wrote:

> In messenger/http_transporter.go
> <https://github.com/mesos/mesos-go/pull/189#discussion_r40300299>:
>
> >  	ch := make(chan error, 1)
> >  	ch <- errTerminal
> > -	return ch
> > +	return upid.UPID{}, ch
>
> I'm not familiar with this code. Why are we returning an empty UPID here
> and the other implementations of this interface below?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesos/mesos-go/pull/189/files#r40300299>.
>